packages/server/src/automations/automationUtils.ts

```diff
@@ -96,6 +98,93 @@ export function getError(err: any) {
   return typeof err !== "string" ? err.toString() : err
 }
 
+export async function sendAutomationAttachmentsToStorage(
+  tableId: string,
+  row: Row
+) {
+  let table = await sdk.tables.getTable(tableId)
+  const attachmentRows: Record<string, AutomationAttachment[]> = {}
+
+  for (const [prop, value] of Object.entries(row)) {
+    const schema = table.schema[prop]
+    if (
+      schema?.type === FieldType.ATTACHMENTS ||
+      schema?.type === FieldType.ATTACHMENT_SINGLE
+    ) {
+      attachmentRows[prop] = value
+    }
+  }
+
+  for (const [prop, attachments] of Object.entries(attachmentRows)) {
+    if (attachments.length) {
+      row[prop] = await Promise.all(
+        attachments.map(attachment => generateAttachmentRow(attachment))
+      )
+    }
+  }
+
+  return row
+}
+
+async function generateAttachmentRow(attachment: AutomationAttachment) {
```

Comment:
This is a _very_ complex function to follow. If possible, it would be great to tackle some of the complexity of it. I appreciate you're going on holiday on Friday, so I don't mind it being a `// TODO` in this PR and then something we tackle in a follow-up PR.

Reply:
I tried to refactor it, how does that look now?

Reply:
Much nicer 🚀